test/suites/integration/subnet_test.go

```diff
@@ -54,6 +54,36 @@ var _ = Describe("Subnets", func() {
 
 		env.ExpectInstance(pod.Spec.NodeName).To(HaveField("SubnetId", HaveValue(Equal(firstSubnet))))
 	})
+	It("should use the subnet-id selector with resource based naming", func() {
+		subnets := getSubnets(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		Expect(len(subnets)).ToNot(Equal(0))
+		availabilityZone := lo.Keys(subnets)
```

Comment:
I think instead of this and the looping below, you could do:

```
allSubnets := lo.Flatten(lo.Values(subnets))
```

which should give you a slice of subnet ID: `[]string{"subnet-id1", "subnet-id2"} 

This is from the lo lib w/ docs here: https://github.com/samber/lo#values